content/200-concepts/100-components/02-prisma-client/052-advanced-type-safety/index.mdx

```diff
@@ -127,4 +127,109 @@ userPosts.posts
 userPosts.profile
 ```
 
-> See the [model query options](/reference/api-reference/prisma-client-reference#model-query-options) reference for more information about the different types available.
+> See the [model query options](../../../../reference/api-reference/prisma-client-reference#model-query-options) reference for more information about the different types available.
+
+### Generated <inlinecode>UncheckedInput</inlinecode> types
+
+The `UncheckedInput` types are a special set of generated types that allow you to perform some operations that Prisma Client considers "unsafe", like directly writing foreign keys. You can choose either the "safe" `Input` types or the "unsafe" `UncheckedInput` type when doing operations like `create`, `update`, or `upsert`.
```

Comment:
I think in Prisma context "foreign keys" are usually referred to as "relation fields", foreign keys are a database level concept.

Reply:
I think the relation field refers to the virtual fields that don't exist in the database (`author`, posts`, etc.). Our docs do refer to the actual foreign keys as "foreign keys" here: https://www.prisma.io/docs/concepts/components/prisma-schema/relations. 

So perhaps this is fine?

Reply:
"relation scalar field" is the proper name for what is the foreign key on SQL/database level: https://www.prisma.io/docs/concepts/components/prisma-schema/relations#relation-scalar-fields

(Many of the uses of "foreign key" on that other page should be replaced by a "relation scalar field" probably)